### PR TITLE
Add option to start router

### PR DESCRIPTION
**Pull request by shakeebkhan66**

### CODE REVIEW

**Comment by om26er** [comment]
@muzzammilshahid can you see if we can reduce the coderabbit npise somehow?

File: lib/screens/mobile/router_dialogbox.dart [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+}
+
+class _RouterDialogBoxState extends State<RouterDialogBox> {
+  final _formKey = GlobalKey<FormState>();
+
+  @override
+  Widget build(BuildContext context) {
+    final realmProvider = context.read<RouterRealmProvider>();
+    final routerProvider = context.read<RouterStateProvider>();
+    final switchProvider = context.read<RouterToggleSwitchProvider>();
```
Comment (muzzammilshahid) [inline lib/screens/mobile/router_dialogbox.dart:25]:
```suggestion
    final realmProvider = Provider.of<RouterRealmProvider>(context, listen: false);
    final routerProvider = Provider.of<RouterStateProvider>(context, listen: false);
    final switchProvider = Provider.of<RouterToggleSwitchProvider>(context, listen: false);
```

File: lib/screens/mobile/router_dialogbox.dart [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+        ),
+      ),
+      validator: (value) {
+        if (value == null || value.isEmpty) {
+          return "Please enter port";
+        }
+        return null;
+      },
+    );
+  }
```
Comment (muzzammilshahid) [inline lib/screens/mobile/router_dialogbox.dart:105]: these two functions seems identical. Can you create a common function for them?

File: lib/screens/mobile/router_dialogbox.dart [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+                  borderRadius: BorderRadius.circular(8),
+                ),
+              ),
+              validator: (value) {
+                if (value == null || value.isEmpty) {
+                  return "Please enter realm here";
+                }
+                return null;
+              },
+            ),
```
Comment (muzzammilshahid) [inline lib/screens/mobile/router_dialogbox.dart:162]:
```suggestion
            title: _buildTextFormField(
              controller: realmProvider.realmControllers[index],
              labelText: "Enter realm here",
              validator: (value) {
                if (value == null || value.isEmpty) {
                  return "Please enter realm here";
                }
                return null;
              },
              keyboardType: TextInputType.text,
            ),
```

**Review by muzzammilshahid — APPROVED** [review]

**Pull request merged**